### PR TITLE
4812 persist terms and conditions data in state

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
per title

### Related Azure Boards Work Items
[AB#4812](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/4812)

### CODE REVIEW

File: frontend/app/routes/public/renew/$id/terms-and-conditions.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { transformFlattenedError } from '~/utils/zod-utils.server';
 
 enum CheckboxValue {
   Yes = 'yes',
+  No = 'no',
```
Comment (nicholas-ly) [inline frontend/app/routes/public/renew/$id/terms-and-conditions.tsx:28]: Nitpick: is `No` needed? I don't see it being used anywhere.

Comment (Fbasham) [reply frontend/app/routes/public/renew/$id/terms-and-conditions.tsx:28]: TS/eslint complains about:
`76:25  error  Unnecessary conditional, both sides of the expression are literal values  @typescript-eslint/no-unnecessary-condition`

There might be another way tho. lemme check

Comment (Fbasham) [reply frontend/app/routes/public/renew/$id/terms-and-conditions.tsx]: `.valueOf()` works, but I hate it for some reason

Comment (nicholas-ly) [reply frontend/app/routes/public/renew/$id/terms-and-conditions.tsx]: No preference either way so I'm resolving.

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**